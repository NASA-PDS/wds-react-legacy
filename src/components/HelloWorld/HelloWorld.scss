.pds-wds-react-hello-world {

   font-size: 1em;

   &.bold {
      font-weight: 700;
   }

   &.heading {
      font-size: 3em;
      font-weight: 700;
   }
   
}