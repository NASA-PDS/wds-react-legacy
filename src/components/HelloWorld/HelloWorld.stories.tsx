import { StoryObj, Meta } from "@storybook/react";
import { HelloWorld } from "./HelloWorld";

export default { 
   component: HelloWorld,
} as Meta<typeof HelloWorld>;
type Story = StoryObj<typeof HelloWorld>;

export const Default: Story = {
   args: {
   },
};

export const CustomName: Story = {
   args: {
     name: "Ceres"
   },
};

export const BoldStyle: Story = {
   args: {
      style: "bold"
   },
};

export const HeadingStyle: Story = {
   args: {
      style: "heading"
   },
};