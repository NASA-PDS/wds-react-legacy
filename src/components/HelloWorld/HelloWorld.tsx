import classNames from "classnames";

import "./HelloWorld.scss";

type HelloWorldProps = {
  /** Name to display in the outputted message */
  name?: string;
  /** Style to use when rendering */
  style?: "bold" | "heading";
  /** Additional classes to apply to component */
  className?: string;
};

/**
 * A basic greeting message displayed to a user.
 *
 * The intent of this component is to showcase how components should be added to our package repository with the use of unit tests, styling, and stories.
 * Default prop values should be specified in the deconstructed parameter object.
 *
 */
export const HelloWorld = ({
  name = "Planetary Data System",
  style,
  className,
  ...otherProps
}: HelloWorldProps) => {

  className = !!className ? className : "";

  const helloWorldClasses = classNames({
    "pds-wds-react-hello-world": true,
    bold: style === "bold",
    heading: style === "heading",
    [className]: !!className,
  });

  return (
    <div className={helloWorldClasses} {...otherProps}>
      Hello, {name}!
    </div>
  );
};