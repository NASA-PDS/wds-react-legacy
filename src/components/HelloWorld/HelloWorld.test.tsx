import renderer from "react-test-renderer"
import { HelloWorld } from "./HelloWorld"

describe('HelloWorld', () => {

  test('HelloWorld component renders correctly', () => {
    const component = renderer.create(
      <HelloWorld />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('The name prop works', () => {
    const component = renderer.create(
      <HelloWorld name={'Universe'} />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

})