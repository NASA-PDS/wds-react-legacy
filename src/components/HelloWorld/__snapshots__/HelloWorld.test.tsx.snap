// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld HelloWorld component renders correctly 1`] = `
<div
  className="pds-wds-react-hello-world"
>
  Hello, 
  Planetary Data System
  !
</div>
`;

exports[`HelloWorld The name prop works 1`] = `
<div
  className="pds-wds-react-hello-world"
>
  Hello, 
  Universe
  !
</div>
`;
